src/mirakuru/unixsocket.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (C) 2019 by Clearcode <http://clearcode.cc>
+# and associates (see AUTHORS).
+
+# This file is part of mirakuru.
+
+# mirakuru is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# mirakuru is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+
+# You should have received a copy of the GNU Lesser General Public License
+# along with mirakuru.  If not, see <http://www.gnu.org/licenses/>.
+"""TCP Socket executor definition."""
+import logging
+import socket
+from typing import Union, List, Tuple, Any
+
+from mirakuru import Executor
+
+log = logging.getLogger(__name__)
+
+
+class UnixSocketExecutor(Executor):
+    """
+    Unixsocket listening process executor.
+
+    Used to start (and wait to actually be running) processes that can accept
+    Unix socket connections.
```

Comment:
Maybe mention that it's for stream sockets only.

At least syslog daemons use datagram sockets (see e.g. that `/dev/log` cannot be checked by the `pre_start_check` here). I don't expect running these to be useful here, while stream sockets are used by many database/queue services useful in tests.

Reply:
True, the datagram are bit different, since they do not upkeep the connection... kind of like tcp and udp from what I've read so far.